rdt/transformers/numerical.py

```diff
@@ -35,15 +37,41 @@ class NumericalTransformer(BaseTransformer):
             If ``True``, always create the new column whether there are null values or not.
             If ``False``, do not create the new column.
             Defaults to ``None``.
+        rounding (int, str or None):
+            Define rounding scheme for data. If set to an int, values will be rounded
+            to that number of decimal places. If ``None``, values will not be rounded.
+            If set to ``'auto'``, the transformer will round to the maximum number of
+            decimal places detected in the fitted data.
     """
 
     null_transformer = None
+    rounding_digits = None
 
-    def __init__(self, dtype=None, nan='mean', null_column=None):
+    def __init__(self, dtype=None, nan='mean', null_column=None, rounding='auto'):
         self.nan = nan
         self.null_column = null_column
         self.dtype = dtype
         self._dtype = dtype
+        self.rounding = rounding
+
+    @staticmethod
+    def _learn_rounding_digits(data):
+        clean_data = data.replace([pd.NA, None], np.nan)
+        if (clean_data % 1 != 0).any():
+            if not (clean_data == clean_data.round(MAX_DECIMALS)).all():
```

Comment:
@csala  I think it is ok to compare like this instead of using `np.allclose` since we aren't rounding past 15 decimals places anymore, but I'm not sure.

Reply:
Yeah, I think so too. Let's keep it like that and make sure that the integration tests validate this thoroughly with random values, so if there are cases in which it fails we will eventually catch it there.